katsdpdata/met_extractors.py

```diff
@@ -109,21 +116,24 @@ def _extract_metadata_from_katdata(self):
         else:
             self.metadata['FileSize'] = str(self._katdata.size)
         self.metadata['KatfileVersion'] = self._katdata.version
-        self.metadata['KatpointTargets'] = [t.description for t in self._katdata.catalogue.targets if t.name not in ['None', 'Nothing']]
+        self.metadata['KatpointTargets'] = [t.description for t in self._katdata.catalogue.targets
+                                            if t.name not in ['None', 'Nothing']]
         self.metadata['NumFreqChannels'] = str(len(self._katdata.channels))
         self.metadata['Observer'] = self._katdata.observer
         self.metadata['RefAntenna'] = self._katdata.ref_ant
-        self.metadata['StartTime'] = time.strftime('%Y-%m-%dT%H:%M:%SZ', time.gmtime(self._katdata.start_time))
-        self.metadata['Targets'] = [t.name for t in self._katdata.catalogue.targets if t.name not in ['None', 'Nothing', 'azel', 'radec']]
-
+        int_secs = floor(self._katdata.start_time.secs)
+        self.metadata['StartTime'] = time.strftime('%Y-%m-%dT%H:%M:%SZ', time.gmtime(int_secs))
+        self.metadata['Targets'] = [t.name for t in self._katdata.catalogue.targets
+                                    if t.name not in ['None', 'Nothing', 'azel', 'radec']]
```

Comment:
I think I created KatpointTarget  because someone requested it. Can't remember who however...

So basically:
- Targets are for archive string searches. (radec/azel are useless as string searches)
- KatpointTarget strings could be used directly to create a katpoint target from the string and radec/azel are valid katpoint targets.

Hope that makes sense?

Reply:
I've added some extra comments.

Reply:
Also, it's a bit sad that Target aliases don't make it into archive search as well, only the first name ;-)

Reply:
Fair enough

Reply:
@ludwigschwardt tell me more :). How can I tease out the target aliases?

Reply:
`t.aliases` is a list of alternate names. Do you expect one name per target or can you handle a bunch of names for search purposes?

Reply:
I guess nowadays folks frequently just put in one name anyway.

Another more general approach would be to use an external lookup eg based on sky position to find alternates. You could do the following in Astropy:
```Python
from astropy.coordinates import SkyCoord
SkyCoord.from_name('PKS 1934-63')
<SkyCoord (ICRS): (ra, dec) in deg
    (294.85427796, -63.71267375)>
```
Then you could do a cone search on the resulting position.

Reply:
Should be able to handle a bunch of names I think.

I guess doing a cone search would put a lot more targets in. Just checking - is this information maybe more relevant for searching catalogues - i.e. is is a good use case for searching for observations?

Reply:
I'm thinking of the reverse use case:

The user comes to the archive and asks for all the observations that pointed at "PKS 1934-63". One option is to scour all names of all targets. Aliases are useful here, because our main calibrator is e.g. "J1939-6342 | PKS1934-638 | PKS 1934-63", with main name "J1939-6342" but "PKS 1934-63" as one of the aliases.

But someone has to remember to add the aliases during observation. More robust is to ask the internet where "PKS 1934-63" is, and then search for observations around that spot in the sky.

Reply:
Would be cool to start adding cone searches, but @ludwigschwardt how does this look for a first solution?  Is it maybe a good plan to also unique the list?
`
target_names = []
for t in self._katdata.catalogue.targets:
    if t.name not in ['None', 'Nothing', 'azel', 'radec']:
        target_names.append(t.name)
    for alias in t.aliases:
        if alias:
            target_names.append(alias)
self.metadata['Targets'] = list(set(target_names))
`

Reply:
That looks good, and unique is helpful. The order tends to be the chronological order of the targets in the file - not sure if you want to preserve that?

Reply:
I don't think it's worth preserving the target order. Search hits would be against string matches.